docs/package-migration-process-guide.md

```diff
@@ -0,0 +1,201 @@
+
+### **[PR#4]** 6. If applicable, rename the migration target package so that it is prepended by the `@metamask/` namespace
+
+- Modify the "name" field in `package.json`.
+- Update the title of the README.md.
+- Add a CHANGELOG entry for the rename.
+
+### **[PR#5]** 7. Create a new release of the migration target from the source repo
+
+- All subsequent releases of the migration target will be made from the core monorepo.
+- [Example PR](https://github.com/MetaMask/eth-json-rpc-provider/pull/29)
+
+## Phase B: **Staging** from the core monorepo's `merged-packages/` directory
+
+### **[PR#6]** 0. Migrate the source repo's git history into the `merged-packages/` temporary directory in core
+
+1. [Install `git-filter-repo`](https://github.com/newren/git-filter-repo/blob/main/INSTALL.md). This tool is like Git's `filter-branch` command, but much easier to use and much less error-prone.
+2. Navigate to a temporary directory: `cd /tmp`
+3. Clone the repo for the library you want to migrate: `git clone https://github.com/MetaMask/<repo-name>` (e.g. `<repo-name>` = `utils`). Do **NOT** use an existing clone.
+4. `cd` into the newly cloned repo.
+5. Run `git filter-repo --to-subdirectory-filter merged-packages/<package-name>` (e.g. `<package-name>` = `utils`). This will rewrite all history to move all files to `merged-packages/<package-name>`. (This is why we're making a fresh clone in a temporary directory — this action is irreversible.)
+6. `cd` to wherever you keep the `core` repo on your machine.
+7. Add the library as a remote: `git remote add <package-name> /tmp/<package-name>`.
+8. Fetch history: `git fetch <package-name> --no-tags`
+9. Make a new branch: `git checkout -b migrate-<package-name>`
+10. Merge the library into the monorepo: `git merge --allow-unrelated-histories <package-name>/<primary-branch>` (e.g. `<primary-branch>` = `main`)
+11. Open a pull request in the core repo that reflects the above changes.
+
+- DO NOT rebase the `migrate-<package-name>` branch.
+- Ensure that superfluous merge commits don't pollute the migrated git history. If necessary, replace the PR branch with a cleaned-up post-migration commit history by rerunning the git history migration steps before merging the PR.
+  - For this, coordinate with the team to minimize the window of time that the PR is open.
+  - For further context on this point: https://github.com/MetaMask/core/pull/1804#issuecomment-1771445829
+- Contact admins to temporarily enable merge commits into main.
+- Merge PR **without squashing** to preserve the migrated git commit history.
+- [Example PR](https://github.com/MetaMask/core/pull/1872)
+
+### 1. Port tags
+
+- Use the `scripts/migrate-tags.sh` tool to port the source repo's release tags into the migrated git history in core Follow these [instructions](https://github.com/MetaMask/core/issues/1800).
+- Port the tags locally with the correct package name prefixes.
+- Push the ported tags to a fork of the core monorepo for testing.
```

Comment:
I think we should highlight the use of a fork for testing. It feels like this could get missed (yes, even though `migrate-tags.md` mentions this).

What do you think about something like:

``` suggestion
- Create a fork of the core monorepo for testing.
- Push the ported tags to a fork of the core monorepo for testing. **Do not run the script against `MetaMask/core` before testing it on a fork.**